File: frontend/lib/types/document.ts
export enum DocumentType {
  BAIL = "bail",
  FACTURE = "facture",
  AVIS_ECHEANCE = "avis_echeance",
  DIAGNOSTIC = "diagnostic",
  RAPPORT_FINANCIER = "rapport_financier",
  AUTRE = "autre",
}

// RAG Source Types - Doit correspondre exactement au backend
export enum SourceType {
  DOCUMENTS = "documents",    // ✅ Correspond au backend
  LEASES = "leases",          // ✅ Correspond au backend
  PROPERTIES = "properties",  // ✅ Correspond au backend
  TENANTS = "tenants",
  KPI = "kpi",
  OWNERS = "owners",
}

export enum FileType {
  PDF = "pdf",
  DOCX = "docx",
  XLSX = "xlsx",
  PPTX = "pptx",
  TXT = "txt",
  CSV = "csv",
  JPG = "jpg",
  JPEG = "jpeg",
  PNG = "png",
  GIF = "gif",
}

export interface Document {
  id: string
  title: string
  description?: string
  document_type: DocumentType
  source_type: SourceType // For RAG indexing
  folder_path: string
  file_path: string
  file_type: FileType
  file_size: number
  organization_id: string
  property_id?: string
  lease_id?: string
  tags: string[]
  uploaded_by: string
  created_at: string
  updated_at: string
  
  // Vectorization fields
  vectorization_status?: "not_planned" | "planned" | "in_progress" | "vectorized" | "error" | "waiting"
  vectorization_started_at?: string
  vectorization_completed_at?: string
  vectorization_error?: string
  qdrant_collection_name?: string
  num_chunks?: number
  content_hash?: string
}

export interface DocumentUploadRequest {
  file: File
  organization_id: string
  title: string
  document_type?: DocumentType
  folder_path?: string
  description?: string
  property_id?: string
  lease_id?: string
  tags?: string[]
}

export interface DocumentUpdateRequest {
  title?: string
  description?: string
  document_type?: DocumentType
  folder_path?: string
  tags?: string[]
  property_id?: string
  lease_id?: string
}

export interface OrganizationQuota {
  organization_id: string
  used_bytes: number
  quota_bytes: number
  used_mb: number
  quota_mb: number
  usage_percentage: number
}

File: AImmoold/backend/app/schemas/document.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from uuid import UUID
from app.core.constants import DocumentType, FileType


class DocumentBase(BaseModel):
    title: str
    description: Optional[str] = None
    document_type: DocumentType = DocumentType.AUTRE
    folder_path: str = "/"
    property_id: Optional[UUID] = None
    lease_id: Optional[UUID] = None
    tags: List[str] = []


class DocumentCreate(DocumentBase):
    organization_id: UUID


class DocumentUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    document_type: Optional[DocumentType] = None
    folder_path: Optional[str] = None
    tags: Optional[List[str]] = None
    property_id: Optional[UUID] = None
    lease_id: Optional[UUID] = None


class Document(DocumentBase):
    id: UUID
    file_path: str
    file_type: FileType
    file_size: int
    organization_id: UUID
    uploaded_by: UUID
    created_at: datetime
    updated_at: datetime
    
    # Vectorization fields
    vectorization_status: Optional[str] = "not_planned"
    vectorization_started_at: Optional[datetime] = None
    vectorization_completed_at: Optional[datetime] = None
    vectorization_error: Optional[str] = None
    qdrant_collection_name: Optional[str] = None
    num_chunks: Optional[int] = 0
    content_hash: Optional[str] = None
    
    class Config:
        from_attributes = True


class DocumentUploadResponse(BaseModel):
    document: Document
    storage_path: str
    public_url: Optional[str] = None


class OrganizationQuota(BaseModel):
    organization_id: UUID
    used_bytes: int
    quota_bytes: int
    used_mb: float
    quota_mb: float
    usage_percentage: float
    
    @property
    def remaining_bytes(self) -> int:
        return max(0, self.quota_bytes - self.used_bytes)
    
    @property
    def is_quota_exceeded(self) -> bool:
        return self.used_bytes >= self.quota_bytes
